camunda-bpm-camel-common/src/main/java/org/camunda/bpm/camel/common/CamelService.java

```diff
@@ -18,9 +18,12 @@
    * @param endpointUri
    *          an Camel {@link org.apache.camel.Endpoint} URI
    *
+   * @exception Exception BpmnError: Raises business error in workflow allowing boundary error handling
```

Comment:
I totally like this change. A concern could be that this changes the behavior (as a BpmError that is not caught terminates a process instance, see note in https://docs.camunda.org/manual/7.12/user-guide/process-engine/delegation-code/#throw-bpmn-errors-from-listeners) and might not be considered backward compatible - WDYT?

Reply:
It is sounding like the BPMError will still end the event if there is no boundary event defined.

So I think this is correct allowing the camel route to throw BPMError to allow the propagation of a business error. All other exceptions (Runtime and Exception) will be treated as a technical error stopping the route as per existing behaviour.

The only difference is a BPMError will not stop the route if there is a boundary event handler. 
I have not tested this **without** a boundary event handler, but I think the BPMError will still stop the route behaving the same as in the past.
So I think this allows adding the boundary event handler as new functionality.

Reply:
I agree that this is really a consistent behavior across the components. It is just a change that is not backwards compatible (if somebody was throwing a BpmnError already, the behavior will change with a new camunda-camel version) which is generally very dangerous to do. I tend to be Ok with it in this case but we should make a note in the readme.